### ISSUE TITLE
After Upgrade to 4.2.0 - got error

### ISSUE THREAD

**Issue by samundra** [lensapp/lens#2438]
**Describe the bug**
Incrementally upgraded to version 4.2.0. After upgrade completed, tried to run application and then got error as shown below.

**Expected behavior**
After upgrade to 4.2.0 expected to work app without errors.

**Screenshots**
![Screen Shot 2021-04-02 at 11 42 58](https://user-images.githubusercontent.com/760855/113381296-d018b100-93a8-11eb-8feb-8e960e4bd098.png)

**Environment (please complete the following information):**
- Lens Version: 4.2.0
- OS: Mojave 10.14.5 (18F132)
- Installation method (e.g. snap or AppImage in Linux): direct .dmg installation

**Logs:**
I have replaced apiUrl with [REDACTED] to hide info.

```
info: 📟 Setting Lens as protocol client for lens://
info: 📟 succeeded ✅
info: 🚀 Starting Lens from "/Users/samundra/Library/Application Support/Lens"
info: 🐚 Syncing shell environment
info: 💾 Loading stores
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-cluster-store.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-workspace-store.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-extensions.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-filesystem-provisioner-store.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-user-store.json
info: 🔑 Getting free port for LensProxy server
info: [CLUSTER-MANAGER]: init cluster {"id":"6afbf7d7-0b03-4f10-87b0-93adf7b1a9ab","name":"uat.test.io","initialized":false,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [CLUSTER-MANAGER]: init cluster {"id":"b1f38fb6-9cb6-44c9-aef3-a0f2c918f370","name":"qa.test.io","initialized":false,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [CLUSTER-MANAGER]: init cluster {"id":"d26e0914-3d7d-4818-84a7-4a8bd3e6e4ee","name":"prod.test.io","initialized":false,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: 🔌 Starting LensProxy
info: [LENS-PROXY]: Proxy server has started at http://localhost:56830
info: 🔎 Testing LensProxy connection ...
info: ⚡ LensProxy connection OK
info: 🖥️  Starting WindowManager
info: 🧩 Initializing extensions
info: [EXTENSION-DISCOVERY] loading extensions from /Users/samundra/Library/Application Support/Lens
info: [CLUSTER]: "uat.test.io" init success {"id":"6afbf7d7-0b03-4f10-87b0-93adf7b1a9ab","context":"uat.test.io","apiUrl":"[REDACTED]"}
info: [CLUSTER]: "qa.test.io" init success {"id":"b1f38fb6-9cb6-44c9-aef3-a0f2c918f370","context":"qa.test.io","apiUrl":"https://api.qa.test.io"}
info: [CLUSTER]: "prod.test.io" init success {"id":"d26e0914-3d7d-4818-84a7-4a8bd3e6e4ee","context":"prod.test.io","apiUrl":"https://api.prod.test.io"}
info: [EXTENSION-INSTALLER] installing dependencies at /Users/samundra/Library/Application Support/Lens
info: [WINDOW-MANAGER]: Loading Main window from url: http://localhost:56830 ...
info: [EXTENSION-INSTALLER] dependencies installed at /Users/samundra/Library/Application Support/Lens
info: [EXTENSION-DISCOVERY] watching extension add/remove in /Users/samundra/.k8slens/extensions
info: [EXTENSION]: enabled lens-license@0.1.0
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/extension-store/lens-survey/preferences-store.json
info: [EXTENSION]: enabled lens-survey@0.1.0
telemetry main extension activated
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/extension-store/lens-telemetry/preferences-store.json
info: [EXTENSION]: enabled lens-telemetry@0.1.0
info: [WINDOW-MANAGER]: Main window loaded
info: 📡 Checking for app updates
info: Checking for update
info: [CLUSTER]: activate {"id":"d26e0914-3d7d-4818-84a7-4a8bd3e6e4ee","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [CLUSTER]: bind events {"id":"d26e0914-3d7d-4818-84a7-4a8bd3e6e4ee","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [CLUSTER]: reconnect {"id":"d26e0914-3d7d-4818-84a7-4a8bd3e6e4ee","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [KUBE-AUTH]: out-channel "kube-auth:d26e0914-3d7d-4818-84a7-4a8bd3e6e4ee" {"data":"Authentication proxy started\n","meta":{"id":"d26e0914-3d7d-4818-84a7-4a8bd3e6e4ee","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}}
info: Update for version 4.2.0 is not available (latest version: 4.2.0, downgrade is disallowed).
warn: PrometheusLens: failed to list services: services "prometheus" not found
warn: PrometheusLens: failed to list services: services "prometheus-server" not found
warn: PrometheusLens: failed to list services: services "prometheus-server" not found
warn: PrometheusLens: failed to list services: services "prometheus" not found
info: [STORE]: SAVING /Users/samundra/Library/Application Support/Lens/lens-cluster-store.json
(node:8048) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.
```
this was working fine before upgrade. No change in kubeconfig.
``

**Comment by samundra** [comment]
I download Lens version `4.1.5` and it works fine without issues.

Result of Lens starting with cli
```
info: 🚀 Starting Lens from "/Users/samundra/Library/Application Support/Lens"
info: 🐚 Syncing shell environment
info: 💾 Loading stores
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/config.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-cluster-store.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-workspace-store.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-extensions.json
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/lens-filesystem-provisioner-store.json
info: 🔑 Getting free port for LensProxy server
info: [CLUSTER-MANAGER]: init cluster {"id":"e418bc5e-2440-4bc2-be5c-391d18ab732e","name":"prod.test.io","initialized":false,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [CLUSTER-MANAGER]: init cluster {"id":"8b2d34f8-db2d-434a-881a-a368608bafb9","name":"uat.test.io","initialized":false,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: 🔌 Starting LensProxy
info: [LENS-PROXY]: Proxy server has started at http://localhost:50937
info: 🔎 Testing LensProxy connection ...
info: [CLUSTER]: "prod.test.io" init success {"id":"e418bc5e-2440-4bc2-be5c-391d18ab732e","context":"prod.test.io","apiUrl":"https://api.prod.test.io"}
info: [CLUSTER]: "uat.test.io" init success {"id":"8b2d34f8-db2d-434a-881a-a368608bafb9","context":"uat.test.io","apiUrl":"[REDACTED]"}
info: ⚡ LensProxy connection OK
info: 🖥️  Starting WindowManager
info: 🧩 Initializing extensions
info: [EXTENSION-DISCOVERY] loading extensions from /Users/samundra/Library/Application Support/Lens
info: [EXTENSION-INSTALLER] installing dependencies at /Users/samundra/Library/Application Support/Lens
info: [WINDOW-MANAGER]: Loading Main window from url: http://localhost:50937 ...
info: [EXTENSION-INSTALLER] dependencies installed at /Users/samundra/Library/Application Support/Lens
info: [EXTENSION-DISCOVERY] watching extension add/remove in /Users/samundra/.k8slens/extensions
info: [EXTENSION]: enabled lens-license@0.1.0
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/extension-store/lens-survey/preferences-store.json
info: [EXTENSION]: enabled lens-survey@0.1.0
telemetry main extension activated
info: [STORE]: LOADED from /Users/samundra/Library/Application Support/Lens/extension-store/lens-telemetry/preferences-store.json
info: [EXTENSION]: enabled lens-telemetry@0.1.0
info: [CLUSTER]: activate {"id":"e418bc5e-2440-4bc2-be5c-391d18ab732e","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [CLUSTER]: bind events {"id":"e418bc5e-2440-4bc2-be5c-391d18ab732e","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [CLUSTER]: reconnect {"id":"e418bc5e-2440-4bc2-be5c-391d18ab732e","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}
info: [KUBE-AUTH]: out-channel "kube-auth:e418bc5e-2440-4bc2-be5c-391d18ab732e" {"data":"Authentication proxy started\n","meta":{"id":"e418bc5e-2440-4bc2-be5c-391d18ab732e","name":"prod.test.io","initialized":true,"ready":false,"online":false,"accessible":false,"disconnected":true}}
info: [WINDOW-MANAGER]: Main window loaded
info: 📡 Checking for app updates
info: Checking for update
info: Found version 4.2.0 (url: Lens-4.2.0-mac.zip, Lens-4.2.0.dmg)
info: [UPDATE-CHECKER]: broadcasting update available {"backchannel":"auto-update:4.2.0","version":"4.2.0"}
warn: PrometheusLens: failed to list services: services "prometheus-server" not found
warn: PrometheusLens: failed to list services: services "prometheus-server" not found
warn: PrometheusLens: failed to list services: services "prometheus" not found
warn: PrometheusLens: failed to list services: services "prometheus" not found
info: [STORE]: SAVING /Users/samundra/Library/Application Support/Lens/lens-cluster-store.json
(node:12790) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
info: [CLUSTER]: refresh {"id":"e418bc5e-2440-4bc2-be5c-391d18ab732e","name":"prod.test.io","initialized":true,"ready":true,"online":true,"accessible":true,"disconnected":false}
```

**Comment by tomharrisonjr** [comment]
I also am having problems post upgrade.  We have 4 clusters: dev works and our new eks cluster works, staging and production are producing an error.  dev, staging, production all on k8s 10.12.  Producing errors in ui:
```
Component stack:


    at Sidebar (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:8493742)
    at aside
    at div
    at IU (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:8498861)
    at AA (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:7676789)
    at t (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:221963)
    at kU (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:8500365)
```

and


```
typeError: Cannot read property '0' of undefined
    at yf.getVersion (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:7872056)
    at yf.getResourceApiBase (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:7871705)
    at http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:8494085
    at Array.map (<anonymous>)
    at http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:8494050
    at Array.map (<anonymous>)
    at BU.get crdSubMenus (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:8493932)
    at ce (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:12079)
    at e.computeValue (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:18380)
    at e.trackAndCompute (http://f807043b-bb69-4d27-acb7-7eee104018d3.localhost:52898/build/Lens.js:2:18206)
```

**Comment by Nokel81** [comment]
Just to confirm, is your CRD `v1` or is it an earlier version?

**Comment by tomharrisonjr** [comment]
> I also am having problems post upgrade. We have 4 clusters: dev works and our new eks cluster works, staging and production are producing an error. dev, staging, production all on k8s 10.12

Update, post 4.2.1 upgrade, still happening (dev is ok, staging, prod are generating the errors noted in my original post).  Not sure I understand why -- the clusters really are nearly identical, generally speaking.  The configs are as well:

**Dev: Working**
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0<redacted>S0t
    server: https://api.us-west-1.dev-2.clearbit.io
  name: us-west-1.dev-2
contexts:
- context:
    cluster: us-west-1.dev-2
    user: tom@clearbit.com
  name: us-west-1.dev-2
current-context: us-west-1.dev-2
kind: Config
preferences: {}
users:
- name: tom@clearbit.com
  user:
    auth-provider:
      config:
        client-id: 0oa22siuvyd8SNFJ42p7
        client-secret: tM<redacted>50
        id-token: eyJ<redacted>olQ
        idp-issuer-url: https://clearbit.okta.com/oauth2/k8s
        refresh-token: rIt<redacted>fCM
      name: oidc
```
**Prod: Fails**
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0<redacted>S0t
    server: https://api.us-west-1.clearbit.io
  name: us-west-1.prod
contexts:
- context:
    cluster: us-west-1.prod
    user: tom@clearbit.com
  name: us-west-1.prod
current-context: us-west-1.prod
kind: Config
preferences: {}
users:
- name: tom@clearbit.com
  user:
    auth-provider:
      config:
        client-id: 0oa2h7xhp6y19QVKf2p7
        client-secret: smR<redacted>nFN
        id-token: eyJ<redacted>cew
        idp-issuer-url: https://clearbit.okta.com/oauth2/k8s
        refresh-token: 0yf<redacted>jzM
      name: oidc
```

Let me know if there's any tracing or logging or other stuff I might be able to provide.

Until then, I guess it's back to previous version :-). 

**And hey: brilliant tool that I have totally come to depend upon -- thanks!!!!**

**Comment by Nokel81** [comment]
This will be fixed in 4.2.2 (not 4.2.1). The reason why it is crashing might be because you are using an older version of kubernetes (only a guess here).

**Comment by samundra** [comment]
I have Lens version 4.2.3. This issue has been fixed in 4.2.3.  I can access dashboard without any errors 

Thank you for the quick solution. 🎉 🎉 🎉 🎉

**Comment by Nokel81** [comment]
You are very welcome.
